File: 백준/Silver/28088. 응애（EASY）/응애（EASY）.cc
#include <iostream>
#include <string.h>
using namespace std;

int main() {
    cin.tie(0)->sync_with_stdio(0);

    int N, M, K; cin >> N >> M >> K;
    bool eungae[N]{0};
    int count[N]{0}, num;

    for(int i=0; i<M; i++) {
        int idx; cin >> idx;
        eungae[idx] = 1;
    }
    num = M;

    while(K--) {
        num = 0;

        for(int i=0; i<N; i++) {
            if(eungae[i] == 1) {
                if(i == 0) {
                    count[N-1]++;
                    count[1]++;
                }
                else if(i == N-1) {
                    count[0]++;
                    count[N-2]++;
                }
                else {
                    count[i-1]++;
                    count[i+1]++;
                }
            }
        }

        for(int i=0; i<N; i++) {
            if(count[i] == 1) {
                eungae[i] = 1;
                num++;
            }
            else eungae[i] = 0; 
        }

        memset(count, 0, sizeof(count));
    }

    cout << num;

    return 0;
}

File: 백준/Silver/28088. 응애（EASY）/응애（EASY）.java
import java.util.*;

public class Main {

    public static void main(String[] args) {
        int N, M, K;
        Scanner sc = new Scanner(System.in);
        N = sc.nextInt();
        M = sc.nextInt();
        K = sc.nextInt();
        sc.nextLine(); 

        int[] eungae = new int[N];
        int[] count = new int[N];

        Arrays.fill(eungae, 0);
        Arrays.fill(count, 0);

        for(int i=0; i<M; i++) {
            int idx;
            idx = Integer.parseInt(sc.nextLine());
            eungae[idx] = 1;
        }

        int num = M;

        for(int i=0; i<K; i++) {
            num = 0;

            for(int j=0; j<N; j++) {
                if(eungae[j] == 1) {
                    if(j == 0) {
                        count[N-1]++;
                        count[1]++;
                    }
                    else if(j == N-1) {
                        count[0]++;
                        count[N-2]++;
                    }
                    else {
                        count[j-1]++;
                        count[j+1]++;
                    }
                }
            }

            for(int j=0; j<N; j++) {
                if(count[j] == 1) {
                    eungae[j] = 1;
                    num++;
                }
                else eungae[j] = 0;
            }

            Arrays.fill(count, 0);
        }

        System.out.println(num);

        sc.close();
    }

}